### PR TITLE
Automatic testing

### PR DESCRIPTION
**Pull request by RomeshA**
Try new PR

### CODE REVIEW

File: atomica/migration.py [hunk 1]
```diff
@@ -1,56 +1,172 @@
 # Central file for migrating Projects
-from distutils.version import LooseVersion
 from .system import logger, AtomicaException
 from .version import version
 import sciris as sc
 from .results import Result
 from .structure import FrameworkSettings as FS
+from distutils.version import LooseVersion
```
Comment (42B) [inline atomica/migration.py]: Isn't this a standard library import? Shouldn't it stay above these others?

* https://pep8.org/#imports

File: atomica/utils.py [hunk 2]
```diff
@@ -44,5 +46,12 @@ def append(self, value):
...
+    def copy(self,old,new):
+        sc.odict.copy(self,old,new)
+        if isinstance(self[new],NamedItem):
+            self[new].name=new
+        return None
 
-
+def parent_dir():
+    # Return the parent directory of the file that called this function
+    return os.path.join(os.path.abspath(os.path.join(inspect.stack()[1][1],os.pardir)),'')
```
Comment (42B) [inline atomica/utils.py]: Newline missing after this line?

File: tests/test_api_workflow.py [hunk 3]
```diff
@@ -1,109 +1,108 @@
...
+scvalues['infdeath']['adults']['y'] = [0.125]
+scvalues['infdeath']['adults']['t'] = [2015.]
+
+scvalues['infdeath']['adults']['y'] = [0.125, 0.5]
+scvalues['infdeath']['adults']['t'] = [2015., 2020.]
+scvalues['infdeath']['adults']['smooth_onset'] = [2, 3]
+
+scvalues['infdeath']['adults']['y'] = [0.125, 0.25, 0.50, 0.50]
+scvalues['infdeath']['adults']['t'] = [2015., 2020., 2025., 2030.]
+scvalues['infdeath']['adults']['smooth_onset'] = [4.,3.,2.,1.]
```
Comment (42B) [inline tests/test_api_workflow.py]:
```python
scvalues = {
    "infdeath": {
        "adults": {
            "y": [0.125, 0.25, 0.5, 0.5],
            "t": [2015.0, 2020.0, 2025.0, 2030.0],
            "smooth_onset": [4.0, 3.0, 2.0, 1.0]
        }
    }
}
```

File: atomica/migration.py [hunk 4]
```diff
@@ -1,56 +1,172 @@
...
+    :return: Updated Project object
+
+    """
+
     migrations = sorted(available_migrations, key=lambda m: LooseVersion(m.original_version))
-    logger.info('Migrating Project "%s" from %s->%s' % (proj.name, proj.version, migrations[-1].new_version))
+    if sc.compareversions(proj.version,version) >= 0:
+        return proj
+    else:
+        logger.info('Migrating Project "%s" from %s->%s' % (proj.name, proj.version, version))
```
Comment (42B) [inline atomica/migration.py]:
```python
logger.info('Migrating Project "%s" from %s->%s', proj.name, proj.version, version)  # ?
```

* https://docs.python.org/3/library/logging.html?highlight=logging#logging.Logger.info

**Comment by RomeshA** [comment]
[Better automation of testing](https://trello.com/c/GcACUo4o/328-better-automation-of-testing)

**Comment by robynstuart** [comment]
ok i think this is ready! looks great!

**Comment by RomeshA** [comment]
[Move example frameworks and databooks out of tests folder?](https://trello.com/c/cnL6JXVi/706-move-example-frameworks-and-databooks-out-of-tests-folder)

**Pull request merged**